app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt

```diff
@@ -56,7 +60,32 @@ class NewsFragment @Inject constructor() : WolmoFragment<NewsPresenter>(), INews
     }
 
     override fun nothingNewToShow() {
-        vSwipeRefreshLayout.isRefreshing = false
         Toast.makeText(context, R.string.nothing_new_to_show, Toast.LENGTH_SHORT).show()
     }
+
+    override fun addRecentNews(recentNews: java.util.ArrayList<News>) {
+        newsDataList.addAll(0, recentNews)
+        viewAdapter.notifyDataSetChanged()
+    }
+
+    override fun addOlderNews(olderNews: java.util.ArrayList<News>) {
+        newsDataList.addAll(newsDataList.size, olderNews)
+        viewAdapter.notifyDataSetChanged()
+    }
+
+    override fun startLoading() {
+        vSwipeRefreshLayout.isRefreshing = true
+    }
+
+    override fun completeLoading() {
+        vSwipeRefreshLayout.isRefreshing = false
+    }
+
+    override fun onLoadOlderNewsError() {
+        Toast.makeText(context, R.string.fail_loading_older_news, Toast.LENGTH_SHORT).show()
```

Comment:
Don't change it, but at Wolmo we have an injectable `ToastFactory` to make Toast usage simpler. Just to let you know.